src/overlays/actors/ovl_En_Hs2/z_en_hs2.c

```diff
@@ -27,25 +29,131 @@ const ActorInit En_Hs2_InitVars = {
+    Actor_SetScale(&this->actor, 0.01f);
+    osSyncPrintf(VT_FGCOL(CYAN) " ヒヨコの店(子人の時) \n" VT_RST);
+    this->actionFunc = func_80A6F1A4;
+    this->unk_2A8 = 0;
+    this->actor.unk_1F = 6;
+}
+
+void EnHs2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHs2* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+s32 func_80A6F0B4(EnHs2* this, GlobalContext* globalCtx, u16 textId, EnHs2ActionFunc actionFunc) {
+    if (func_8002F194(&this->actor, globalCtx)) {
+        this->actionFunc = actionFunc;
+        return 1;
+    }
+
+    this->actor.textId = textId;
+    if (ABS((s16)(this->actor.yawTowardsLink - this->actor.shape.rot.y)) < 0x2151 &&
+        this->actor.xzDistFromLink < 100.0f) {
+        this->unk_2A8 |= 0x1;
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    }
+    return 0;
+}
+
+void func_80A6F164(EnHs2* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx)) {
+        this->actionFunc = func_80A6F1A4;
+    }
+    this->unk_2A8 |= 0x1;
+}
+
+void func_80A6F1A4(EnHs2* this, GlobalContext* globalCtx) {
+    u16 textId;
+
+    textId = Text_GetFaceReaction(globalCtx, 9);
+    if (textId == 0) {
+        textId = 0x5069;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hs2/func_80A6F0B4.s")
+    func_80A6F0B4(this, globalCtx, textId, func_80A6F164);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hs2/func_80A6F164.s")
+void EnHs2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnHs2* this = THIS;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hs2/func_80A6F1A4.s")
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        this->skelAnime.animCurrentFrame = 0.0f;
+    }
+    this->actionFunc(this, globalCtx);
+    if (this->unk_2A8 & 0x1) {
+        func_80038290(globalCtx, &this->actor, &this->unk_29C, &this->unk_2A2, this->actor.posRot2.pos);
+        this->unk_2A8 &= 0xFFFE;
+    } else {
+        Math_SmoothScaleMaxMinS(&this->unk_29C.x, 12800, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_29C.y, 0, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_2A2.x, 0, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_2A2.y, 0, 6, 6200, 100);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hs2/EnHs2_Update.s")
+s32 EnHs2_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* pos, Vec3s* rot, Actor* thisx) {
+    EnHs2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hs2/func_80A6F358.s")
+    switch (limbIndex) {
+        case 12:
+        case 13:
+            *dList = NULL;
+            return 0;
```

Comment:
Can all those returns be breaks instead?

Reply:
It doesn't match without them